doc/1.4/language.md

```diff
@@ -3815,6 +3828,271 @@ independent method callback(int i, void *aux) {
+    method write_allowed() -> (bool) default {
+        return true;
+    }
+
+    method alter_write(uint64 curr, uint64 written) -> (uint64) default {
+        return write_allowed() ? written : curr;
+    }
+}
+
+template write_1_clears is alter_write {
+    method alter_write(uint64 curr, uint64 written) -> (uint64) default {
+        return curr & ~written;
+    }
+}
+
+template gated_write_1_clears is (gated_write, write_1_clears) {
+    method alter_write(uint64 curr, uint64 written) default {
+        local uint64 new = this.templates.write_1_clears.alter_write(
+            curr, written);
+        return this.templates.gated_write.alter_write(curr, new);
+    }
+}
+
+// Resolve the conflict introduced whenever the two orthogonal templates are
+// instantiated by also instantiating gated_write_1_clears when that happens
+in each (gated_write, write_1_clears) { is gated_write_1_clears; }
+```
+
+The following is an example of the second case:
+```
+template very_complex_register is register {
+    method write_register(uint64 written, uint64 enabled_bytes,
+                          void *aux) default {
+        ... // An extremely complicated implementation
+    }
+}
+
+template gated_register is register {
+    method write_allowed() -> (bool) default {
+        return true;
+    }
+
+    method on_write_attempted_when_not_allowed() default {
+        log spec_viol: "%s was written to when not allowed", qname;
+    }
+
+    method write_register(uint64 written, uint64 enabled_bytes,
+                          void *aux) default {
+        if (write_allowed()) {
+            default(written, enabled_bytes, aux);
+        } else {
+            on_write_attempted_when_not_allowed();
+        }
+    }
+}
+
+template very_complex_gated_register is (very_complex_register,
+                                         gated_register) {
+    // No sensible way to combine the two implementations by calling both.
+    // Even if there were, calling both implementations would cause each field
+    // of the register to be written to multiple times, potentially duplicating
+    // side-effects, which is undesirable.
+    // Instead, very_complex_register is chosen as the base implementation
+    // called, and the behaviour of gated_register is replicated around that
+    // call.
+    method write_register(uint64 written, uint64 enabled_bytes,
+                          void *aux) default {
+        if (write_allowed()) {
+            this.templates.very_complex_register.write_register(
+                written, enabled_bytes, aux);
+        } else {
+            on_write_attempted_when_not_allowed();
+        }
+    }
+}
+
+in each (gated_register, very_complex_register) {
+    is very_complex_gated_register;
+}
+```
+
+A template-qualified method implementation call is resolved by using
+the method implementation provided to the object by the named template.
+If no such implementation is provided (whether it be because the template does
```

Comment:
Either `whether it is because` or `be it because`, I think

Reply:
A googling says "whether it be" is correct grammar. Your options would also work (though _maybe_ not `whether it is because`? My English skills are failing me here)

Reply:
ok, then the lack of English skills is probably on my side.